poh/src/poh_recorder.rs

```diff
@@ -646,30 +652,35 @@ impl PohRecorder {
                 return Err(PohRecorderError::MaxHeightReached);
             }
 
-            {
+            let mut poh_lock = {
                 let now = Instant::now();
-                let mut poh_lock = self.poh.lock().unwrap();
-
+                let poh_lock = self.poh.lock().unwrap();
                 self.record_lock_contention_us += timing::duration_as_us(&now.elapsed());
+                poh_lock
+            };
+
+            let record_res = {
                 let now = Instant::now();
                 let res = poh_lock.record(mixin);
-                drop(poh_lock);
                 self.record_us += timing::duration_as_us(&now.elapsed());
+                res
+            };
+
+            drop(poh_lock);
+
+            if let Some(poh_entry) = record_res {
                 let now = Instant::now();
-                if let Some(poh_entry) = res {
-                    let entry = Entry {
-                        num_hashes: poh_entry.num_hashes,
-                        hash: poh_entry.hash,
-                        transactions,
-                    };
-                    let bank_clone = working_bank.bank.clone();
-                    self.prepare_send_us += timing::duration_as_us(&now.elapsed());
```

Comment:
Didn't feel like this granularity was needed because Entry creation and bank clone is cheap?